test/pg-cdc/constraints.td

```diff
@@ -0,0 +1,88 @@
+INSERT INTO pk_table VALUES (2, 'two');
+
+INSERT INTO pk_table VALUES (3, 'three');
+
+CREATE TABLE nonpk_table (f1 INTEGER, f2 INTEGER);
+INSERT INTO nonpk_table VALUES (1, 1), (1, 1);
+ALTER TABLE nonpk_table REPLICA IDENTITY FULL;
+INSERT INTO nonpk_table VALUES (2, 2), (2, 2);
+
+CREATE TABLE complex_constraints (pk INTEGER, f2 TEXT, UNIQUE (f2), UNIQUE (f2, pk));
+ALTER TABLE complex_constraints ADD PRIMARY KEY (pk);
+INSERT INTO complex_constraints VALUES (1, 'one');
+ALTER TABLE complex_constraints REPLICA IDENTITY FULL;
+INSERT INTO complex_constraints VALUES (2, 'two');
+INSERT INTO complex_constraints VALUES (3, 'three');
+
+> CREATE SOURCE mz_source
+  FROM POSTGRES CONNECTION pg (PUBLICATION 'mz_source')
+  FOR TABLES (pk_table, nonpk_table, complex_constraints);
+
+> CREATE DEFAULT INDEX ON pk_table;
+
+# Default index only includes primary key column
+> SHOW CREATE INDEX pk_table_primary_idx;
+name                                     create_sql
+-------------------------------------------------------------------------------------------------------------------------------------------------------
+materialize.public.pk_table_primary_idx "CREATE INDEX \"pk_table_primary_idx\" IN CLUSTER \"default\" ON \"materialize\".\"public\".\"pk_table\" (\"pk\")"
+
+# Not null constraint optimizes like this:
+> EXPLAIN SELECT * FROM pk_table WHERE pk IS NULL;
+"Explained Query (fast path):\n  Constant <empty>\n"
+
+# Default index includes all columns
+> CREATE DEFAULT INDEX ON nonpk_table;
+
+# Default index only includes primary key column
+> SHOW CREATE INDEX nonpk_table_primary_idx;
+name                                     create_sql
+-------------------------------------------------------------------------------------------------------------------------------------------------------
+materialize.public.nonpk_table_primary_idx "CREATE INDEX \"nonpk_table_primary_idx\" IN CLUSTER \"default\" ON \"materialize\".\"public\".\"nonpk_table\" (\"f1\", \"f2\")"
+
+# No not-null constraint
+> EXPLAIN SELECT * FROM nonpk_table WHERE f1 IS NULL;
+"Explained Query (fast path):\n  Filter (#0) IS NULL\n    ReadExistingIndex materialize.public.nonpk_table_primary_idx\n\nUsed Indexes:\n  - materialize.public.nonpk_table_primary_idx\n"
+
+> CREATE DEFAULT INDEX ON complex_constraints;
+
+# Default index only includes primary key column
+> SHOW CREATE INDEX complex_constraints_primary_idx;
+name                                     create_sql
+-------------------------------------------------------------------------------------------------------------------------------------------------------
+materialize.public.complex_constraints_primary_idx "CREATE INDEX \"complex_constraints_primary_idx\" IN CLUSTER \"default\" ON \"materialize\".\"public\".\"complex_constraints\" (\"pk\")"
+
+# Not null constraint optimizes like this:
+> EXPLAIN SELECT * FROM pk_table WHERE pk IS NULL;
+"Explained Query (fast path):\n  Constant <empty>\n"
```

Comment:
A query with `DISTINCT` is needed to confirm that the `pk` is considered unique and no de-duplication will be performeed:

```
materialize=> explain select distinct pk from pk_table;
                        Optimized Plan                         
---------------------------------------------------------------
 Explained Query (fast path):                                 +
   Project (#0)                                               +
     ReadExistingIndex materialize.public.pk_table_primary_idx+
                                                              +
 Used Indexes:                                                +
   - materialize.public.pk_table_primary_idx                  +
```

Testdrive also has the `?` operator that allows multi-line explains without \n and quotes, like this:

```
? EXPLAIN SELECT * FROM t1 AS a1, t1 AS a2 WHERE a1.f1 IS NOT NULL;
Explained Query:
  CrossJoin type=differential
    ArrangeBy keys=[[]]
      Filter (#0) IS NOT NULL
        Get materialize.public.t1
    ArrangeBy keys=[[]]
      Get materialize.public.t1
```

Reply:
ty for this pointer--used this thoroughly in the tests I added.

I could not get `?` to work with `EXPLAIN SELECT`'s output, it kept fussing at me about some mismatched whitespaces; glad to dig into this but was quicker to just use the `\n` version to get this PR out.